src/ORM/Rule/ExistsIn.php

```diff
@@ -125,6 +126,51 @@ public function __invoke(EntityInterface $entity, array $options)
             }
         }
 
+        if (isset($options['parent']) && is_a($options['parent'], EntityInterface::class)) {
+            /**
+             * @var EntityInterface $parent
+             */
+            $parent = $options['parent'];
+
+            //extract entity from given parent regarding HM and HABTM associations
+            $parentEntity = null;
+            if ($isAssociation) {
+                if (TableRegistry::get($parent->getSource()) === $target->getTarget()) {
+                    //parent is of same  as target
+                    $parentEntity = $parent;
+                } else {
+                    if (!empty($parent[strtolower($target->getName())])) {
+                        $parentEntity = $parent[strtolower($target->getName())];
```

Comment:
You should use the association's `getProperty()` here instead. The default conventions use `lower_case_underscore` for multi-word properties which this approach won't handle well.